### PR TITLE
Change to new calling and py3 object for modules

### PR DESCRIPTION
**Pull request by tobes**
**This is outdated see comment below**

This PR follows on from discussions in issue #250  I'd like to thank @cornerman for the `self.py3` contribution.

What this PR does:

Any module that implements `class Meta` is treated specially.

* Calls to its `on_click` and update methods are called simply as `update()` and `on_click(event)` rather than the old style `update(i3s_output_list, i3s_config, event)` and `on_click(i3s_output_list, i3s_config, event)`.

* A special helper object is injected into the module as `self.py3`.  This helper module includes: 

  -  helper functions:

    - `notify_user()` send message to user.

    - `get_module_info()` get info about other modules (`group` module  needs to do this).

    - `trigger_event()` trigger an event for a module.

    - `update()` force update of self or another module.

  - constant:

    - `CACHE_FOREVER` if returned as `cache_until` causes the module to not be scheduled for an update (events would still trigger an update)

So basically it allows modules to be written in a cleaner way and have easy access to extra functionality.

here is `fish.py` an example module
```
# -*- coding: utf-8 -*-

class Py3status:

    class Meta:
        pass

    def fish(self):
        return {
            'full_text': '>(((º>',
            'cached_until': self.py3.CACHE_FOREVER
        }

    def on_click(self, event):
        self.py3.notify_user('くコ:彡')
```

Additionally the `group` module is updated

### CODE REVIEW

**Comment by tobes** [comment]
This is an updated version of the patch based on feedback from @ultrabug 

`class Meta` has been removed.

All modules can now define `update(self)` or `update(self, i3s_output_list, i3s_config)` they will be called appropriately.  same for `on_click()` and `kill()`.

There can even be a mix of old and new calling methods.

`self.py3` is injected into any module that does not have it defined as an attribute.

Example updated:

```
# -*- coding: utf-8 -*-

class Py3status:

    def fish(self):
        return {
            'full_text': '>(((º>',
            'cached_until': self.py3.CACHE_FOREVER
        }

    def on_click(self, event):
        self.py3.notify_user('くコ:彡')
```

**Comment by pydsigner** [comment]
This will make exposing logging to modules so much easier.

**Comment by ultrabug** [comment]
What can I say, it's very elegant, congratulations :palm_tree:

**Comment by tobes** [comment]
A big thanks to @ultrabug and @cornerman on this.  You both really helped make this with your suggestions. :+1:
